main/src/ca/uwaterloo/flix/api/lsp/Visitor.scala

```diff
@@ -0,0 +1,937 @@
+
+      case Expr.FixpointSolve(exp, _, _, _) =>
+        visitExpr(exp)
+
+      case Expr.FixpointFilter(_, exp, _, _, _) =>
+        visitExpr(exp)
+
+      case Expr.FixpointInject(exp, _, _, _, _) =>
+        visitExpr(exp)
+
+      case Expr.FixpointProject(_, exp, _, _, _) =>
+        visitExpr(exp)
+
+      case Expr.Error(_, _, _) => ()
+    }
+  }
+
+  private def visitSigSymUse(symUse: SigSymUse)(implicit a: Acceptor, c: Consumer): Unit = {
+    val SigSymUse(_, loc) = symUse
+    if (!a.accept(loc)) { return }
+
+    c.consumeSigSymUse(symUse)
+  }
+
+  private def visitLocalDefSymUse(symUse: LocalDefSymUse)(implicit a: Acceptor, c: Consumer): Unit = {
+    val LocalDefSymUse(_, loc) = symUse
+    if (!a.accept(loc)) { return }
+
+    c.consumeLocalDefSym(symUse)
+  }
+
+  private def visitPredicateParam(pparam: PredicateParam)(implicit a: Acceptor, c: Consumer): Unit = {
+    val PredicateParam(_, _, loc) = pparam
+    if (!a.accept(loc)) { return }
+
+    c.consumePredicateParam(pparam)
+  }
+
+  private def visitStructFieldSymUse(symUse: StructFieldSymUse)(implicit a: Acceptor, c: Consumer): Unit = {
+    val StructFieldSymUse(_, loc) = symUse
+    if (!a.accept(loc)) { return }
+
+    c.consumeStructFieldSymUse(symUse)
+  }
+
+  private def visitEffectSymUse(effUse: EffectSymUse)(implicit a: Acceptor, c: Consumer): Unit = {
+    val EffectSymUse(_, loc) = effUse
+    if (!a.accept(loc)) { return }
+
+    c.consumeEffectSymUse(effUse)
+  }
+
+  private def visitJvmMethod(method: JvmMethod)(implicit a: Acceptor, c: Consumer): Unit = {
+    val JvmMethod(_, fparams, exp, retTpe, _, loc) = method
+    if (!a.accept(loc)) { return }
+
+    c.consumeJvmMethod(method)
+
+    fparams.foreach(visitFormalParam)
+    visitExpr(exp)
+    visitType(retTpe)
+  }
+
+  private def visitDefSymUse(symUse: DefSymUse)(implicit a: Acceptor, c: Consumer): Unit = {
+    val DefSymUse(_, loc) = symUse
+    if(!a.accept(loc)) { return }
+
+    c.consumeDefSymUse(symUse)
+  }
+
+  private def visitSelectChannelRule(rule: SelectChannelRule)(implicit a: Acceptor, c: Consumer): Unit = {
+    val SelectChannelRule(_, chan, exp) = rule
+    // TODO `insideRule` is a hack, should be removed eventually. Necessary for now since SelectChannelRule don't have locations
+    val insideRule = a.accept(chan.loc) || a.accept(exp.loc)
+    if (!insideRule) { return }
+
+    c.consumeSelectChannelRule(rule)
+
+    visitExpr(chan)
+    visitExpr(exp)
+  }
+
+  private def visitFormalParam(fparam: FormalParam)(implicit a: Acceptor, c: Consumer): Unit = {
+    val FormalParam(_, _, tpe, _, loc) = fparam
+    if (!a.accept(loc)) { return }
+
+    c.consumeFormalParam(fparam)
+
+    visitType(tpe)
+  }
+
+  private def visitHandlerRule(rule: HandlerRule)(implicit a: Acceptor, c: Consumer): Unit = {
+    val HandlerRule(op, fparams, exp) = rule
+    // TODO `insideRule` is a hack, should be removed eventually. Necessary for now since HandlerRules don't have locations
```

Comment:
Does #8839 fix this? Otherwise I suppose this can be a follow-up PR.

Reply:
Same as below

Reply:
Roger. Follow-up  PR.